### PR TITLE
Adding Logging Enabled Check to Azure Key Vault

### PR DESCRIPTION
**Pull request by jnikles1**
Signed-off-by: Justin Nikles <justin.nikles@sap.com>

### Description

Adds a check specifically for the AuditEvent log setting, which is set to true when logging is enabled https://docs.microsoft.com/en-us/azure/key-vault/general/howto-logging?tabs=azure-cli#enable-logging

### Issues Resolved

List any existing issues this PR resolves, or any Discourse or StackOverflow discussion that's relevant

### Check List

- [N/A] New functionality includes tests
- [x] All tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Comment by jnikles1** [comment]
the diagnostic_settings_logs method doesn't sufficiently report whether logging is enabled or not. If any setting in the log block that is returned is not enabled (whether it signals that logging is enabled or not) the array reports a false flag. For this reason, the additional method was added to specifically check the field which signals that logging is enabled.

File: libraries/azure_key_vault.rb [hunk 1]
```diff
@@ -99,6 +99,18 @@ def diagnostic_settings_logs
     end
     result
   end
+
+  def has_logging_enabled?
+    return false if diagnostic_settings.nil? || diagnostic_settings.empty?
+    diagnostic_settings.each do |setting|
+      logs = setting.properties&.logs
+      next unless logs
+      logs.each do |log|
```
Comment (sathish-progress) [inline libraries/azure_key_vault.rb:108]: this block will return `[]` when `log.category` is not an `AuditEvent`

File: libraries/azure_key_vault.rb [hunk 2]
```diff
@@ -99,6 +99,18 @@ def diagnostic_settings_logs
...
+        next unless log.category == 'AuditEvent'
+        return log.enabled
+      end
+    end
```
Comment (sathish-progress) [inline libraries/azure_key_vault.rb:112]:
```suggestion
    diagnostic_settings.each do |setting|
      log = setting.properties&.logs&.detect{|log| log.category == 'AuditEvent' }
      return log&.enabled
    end
```

**Comment by jnikles1** [comment]
hi @sathish-progress - thanks for the suggestion. It's now implemented as suggested

**Review by SathishAchilles — APPROVED** [review]
LGTM

**Review by sathish-progress — APPROVED** [review]

**Pull request merged**